qiime2/tests/test_artifact_api.py

```diff
@@ -118,6 +118,20 @@ def test_reload_fails(self):
         with self.assertRaises(ImportError):
             importlib.reload(qiime2.plugins.dummy_plugin)
 
+    def test_import_base_format_stores_checksums(self):
+        from qiime2 import Artifact
+        from qiime2.core.testing.type import IntSequence2
+        from qiime2.core.testing.format import IntSequenceFormat
+        from qiime2.core.util import load_action_yaml
+        from qiime2.plugin.util import transform
+
+        ff = transform([1, 2, 3,], to_type=IntSequenceFormat)
+        ff2 = Artifact.import_data(IntSequence2, ff,
+                                   view_type='IntSequenceFormat')
```

Comment:
I think this line needs to be a class (instead of the name as a string) in order for the original bug to appear.

You can verify by "un-fixing" the PR and verifying the test fails.